test/test_utils.cpp

```diff
@@ -386,3 +387,67 @@ TEST(ModelReading, equations)
         EXPECT_DOUBLE_EQ(node.x(), node.y());
 }
 
+TEST(Quaternion, addition)
+{
+    biorbd::utils::Quaternion q1(1,2,3,4);
+    biorbd::utils::Quaternion q2(2,3,4,5);
+    EXPECT_NEAR(q1[3], 1, requiredPrecision);
```

Comment:
Hum... Does it feels like a bug to you (In a user perspective I mean) that the components shuffle around? If you think so, I can take it from there to make it consistent

Reply:
It is a matter of convention. In rbdl they are defined as (x,y,z,w). But in the biorbd constructors, we use multiple logics. I.E : (w,x,y,z), or directly a vector which will be interpreted in rbdl as (x,y,w,z). Why not using a single constructor which would call the rbdl default one (using whatever convention, I personally prefer (w,x,y,z)), and then all the other constructors should refer to this one, in order to limit the possible confusions.
The reason why the test is performed in this order is arbitrary though.